src/sentry/search/events/fields.py

```diff
@@ -219,9 +220,15 @@ def team_key_transaction_expression(organization_id, team_ids, project_ids):
         .distinct("transaction", "project_id")
     )
 
-    # There are team key transactions marked, so hard code false into the query.
-    if len(team_key_transactions) == 0:
+    count = len(team_key_transactions)
+
+    # There are no team key transactions marked, so hard code false into the query.
+    if count == 0:
         return ["toInt8", [0]]
+    elif count > MAX_QUERYABLE_TEAM_KEY_TRANSACTIONS:
+        raise InvalidSearchQuery(
```

Comment:
This is only thrown if teams sent in the query have under 500 key transactions? So this won't leak into existing requests until a ton of key transactions have been made, right?

Reply:
This only affects the `team_key_transaction` field which is not available in the UI, and will be feature gated.

Additionally, this will result in a 400 error that we have a nice error banner to display this message.